microhapulator/tests/test_pipe.py

```diff
@@ -39,9 +39,8 @@ def test_pipe_missing_files(tmp_path):
 
 
 def test_pipe_gbr_usc10(tmp_path):
-    hg38 = str(tmp_path / "hg38-placeholder.fasta")
-    copyfile(data_file("refr/usc10-refr.fna"), hg38)
-    run(["bwa", "index", hg38])
+    hg38_placeholder = str("microhapulator/data/hg38.fasta")
+    copyfile(data_file("refr/usc10-refr.fna"), hg38_placeholder)
```

Comment:
Now that we're downloading the GRCh38 reference as part of the CI build, I think we can just drop the placeholder approach for this test. Certainly I don't think we should overwrite the installed genome file with the small reference file, as this seems to be doing...?

Reply:
I kept this here initially because I saw that if the index was present but the full reference genome was not, the genome was getting downloaded and re-indexed so this was a work around to avoid that, but a better solution is change to the code to only index if the index isn't already there. I tracked this issue down to the download_full_reference Snakemake rule and found that if one of the expected outputs (like the reference genome) wasn't present, all of the outputs that were already there (the index) were getting deleted and then getrefr was recreating them since they were no longer there. This seems weird to me but maybe there is a reason for this behavior of Snakelike? My fix for this was changing the input of map_full_reference to be the index instead of the genome, so that if the index is already there it skips straight to that step. I also removed the reference genome as an expected output of download_full_reference and changed it to just the index files. This means that now if the genome is present but it hasn't been index files are not, the genome fasta will not be unnecessarily deleted and if the genome is not present, getrefr contains the logic to download and then index it.  TL;DR now during the CI, we can download just the index and not the full reference genome without having to use this placeholder.

Reply:
Ok, I may have been thrown of by the assumption that `bwa mem` uses the original FASTA file in addition to the index files. If it does not, then the code referenced above is merely *misleading* rather than technically *incorrect*. :smiley:

In any case, I think what you've done in the most recent commits makes sense. Since we're downloading the reference for CI anyway, just use that. It certainly makes the test code more concise!